### PR TITLE
input arrays treated as single value unless explicitly mapped using $.

### PR DESCRIPTION
**Pull request by andrew-coleman**
This is a proposed solution to bugs #332 and #316 which I think also addresses the comments that were discussed in #170 

This change only affects how top level JSON arrays are handled in the input data.  If you wish to explicitly map the JSONata expression to _each_ item in the input array, then the expression will need to explicitly declare that by starting it with `$.`  Otherwise the input array as a whole will be treated as the single context value of the expression.

I would argue that this is not a breaking change because the current behaviour is somewhat broken (hence the bug reports).  Although I have had to modify a couple of test results (edge cases, I think).

### CODE REVIEW

File: src/jsonata.js [hunk 1]
```diff
@@ -1904,6 +1904,12 @@ var jsonata = (function() {
                 timestamp = new Date();
                 exec_env.timestamp = timestamp;
 
+                // if the input is a JSON array, then wrap it in a singleton sequence si it gets treated as a single input
```
Comment (mattbaileyuk) [inline src/jsonata.js]: Minor typo: `si` -> `so`
